### ISSUE TITLE
parse keepalive as a seconds deltatime

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1924]
currently it is treated as an integer and assumed to be seconds

**Comment by letoams** [comment]
As long as it doesn’t allow 0 < x < 1

**Comment by cagney** [comment]
> As long as it doesn’t allow 0 < x < 1

I'm curious, why?

**Comment by letoams** [comment]
On Sun, 17 Nov 2024, cagney wrote:

>       As long as it doesn’t allow 0 < x < 1
> 
> I'm curious, why?

sending keepalives faster than 1 per second is insane. When the
units were seconds, this was not possible. Now the units are ms,
it could be done. I think it should be prevented. Arguably, maybe
even a lower limit of 10s would make sense.

**Comment by cagney** [comment]
> sending keepalives faster than 1 per second is insane

I agree, but how much cotton wool should pluto wrap the user in?  Issue a stern warning but allow it.
It feels like we keep adding checks, only to then need to add an impair so the check can be undone.

Either way, it needs a test.

**Comment by cagney** [comment]
strangely, the pluto option `--keep-alive` gets this check:
```

			if (deltatime_cmp(keep_alive, >, deltatime(secs_per_day))) {
				fatal_opt(longindex, logger, "exceeds 1 day");
			}
```
but not `keep-alive=2d` in the config file

**Comment by cagney** [comment]
```
grep keep.*alive= testing/pluto/*/*.conf
testing/pluto/ikev2-mobike-08-nat-expire-liveness-east/ipsec.conf:	nat-keepalive=no
```
sigh

**Comment by cagney** [comment]
It's ignoring <1s >1d for now.  Can pick this up after 5.2.
